mesa/agent.py

```diff
@@ -57,3 +60,41 @@ def advance(self) -> None:
     @property
     def random(self) -> Random:
         return self.model.random
+
+
+class AgentSet:
+    def __init__(self, agents: set[Agent], model: Model):
+        self._agents = agents
+        self.model = model
+
+    def __len__(self):
+        return len(self._agents)
+
+    def __iter__(self) -> Iterator[Agent]:
+        return iter(self._agents)
+
+    def __contains__(self, agent: Agent) -> bool:
+        """Check if an agent is in the AgentSet."""
+        return agent in self._agents
+
+    def select(self, filter_func: Callable[[Agent], bool] | None = None) -> AgentSet:
+        if filter_func is None:
+            return AgentSet(set(self._agents), self.model)
+        return AgentSet(
+            {agent for agent in self._agents if filter_func(agent)}, self.model
+        )
+
+    def shuffle(self) -> AgentSet:
+        shuffled_agents = list(self._agents)
+        self.model.random.shuffle(shuffled_agents)
+        return AgentSet(set(shuffled_agents), self.model)
```

Comment:
by turning it into a set, you loose the ordering again....

Reply:
Good catch, thanks for fixing!